Dev fall2020

jolin0504 opened the pull request:
devfall2020 pull request, devsum2020 closed

jolin0504 committed:
Push to Fall Dev Branch

jolin0504 committed:
update

jolin0504 committed:
still needed fixing

jolin0504 committed:
last updated 01/08
still some text not fixed, icon, textinput

neilhatfield committed:
Switch to Single File

This update switches from the two-file to the single file (app.r) system.

neilhatfield reviewed (dismissed):
I've switched the app over to the single file format as well as a few other changes. You'll need to pull this version of the code before you make any changes.

I'm getting the following errors when I run the app:

- No graphs are displayed initially on the explore page
- When the Refresh data button is clicked the I get the message `Warning: Error in $: object of type 'builtin' is not subsettable` with a reference to Line 780
- When the Generate New Model button is clicked, the app crashes with the message `Warning: Error in <-: object 'value' not found`

The above errors need to be explored and resolved. One potential issue for first error message could be a change in the `regsubsets` behavior. Investigation is needed.

jolin0504 committed:
misspelling mistake

jolin0504 committed:
Merge branch 'devFall2020' of github.com:EducationShinyAppTeam/Variable_Selection into devFall2020

# Conflicts:
#	ui.R

ymy5178 reviewed (dismissed):
Recommended Priority: High

1. There is no description file.
2. Readme file does not meet the requisites of the current Style Guide.
3. One issue that should round to more decimals on the y-axis has not been resolved.
4. Problems in graphs of model selection in Exploring Real Data. There should include five Xs, but there are only two which would make users cannot get the right answer in Check Yourself.
![image](https://user-images.githubusercontent.com/70001189/171094985-c8b44218-dcfa-4496-b792-22f0141aa6c8.png)
5. The button of describing the best estimated model is always chosen even after I refresh data when I choose it for one time. This made me have to make sure that the box is not chosen before I refresh data. Otherwise, it would be very easy for me to see the right answer.
![image](https://user-images.githubusercontent.com/70001189/171095540-2f47fa70-4496-4dfa-b769-621cb54390a6.png)
6. Explanation is needed. Even I took several trials, for some criterions, I still cannot know why the best estimated model is that one exactly. Therefore, I think explanation is needed.

ymy5178 committed:
description and readme

add description and readme file to the folder.
create docs folder and add the screenshot.

ymy5178 committed:
fix all, it can be reviewed now i think

fix mismatched parentheses and add the definition of value and c

ymy5178 committed:
last update

change the last update

ymy5178 committed:
style guide

changes made in style guide

Maxinewsu reviewed (dismissed):
I think this app is good to go.

JuddHe reviewed (dismissed):
I think it is good.

neilhatfield commented:
@ymy5178, @Maxinewsu, and @JuddHe 
I'm dismissing the reviews as there are several issues that should have been caught in peer reviewing. Please remember to use the [Reviewing Checklist app](https://psu-eberly.shinyapps.io/Reviewing_Checklists/) to help ensure that you check all major aspects.
### DESCRIPTION file
- [x] Incorrect Authors: Zhiro Wang is the author; and Zhoulin Luo and Yijun Yao as contributors (ctb)
- [x] Set the date to 2019-08-01
- [x] The learning objectives are incorrect
- [x] The description should be expanded (see later comments)
### README file
- [x] Remove the new badge from the badge list
- [x] The year should be 2019
- [x] Expand the description of the app to be more than the current extremely short sentence.
### Delete Extraneous Files
- [x] Xserver.R
- [x] Xui.R
- [x] manifest.json
- [x] In the `www` folder, both Feature.css and logo.png
### App.R
- [x] Reformat lines so that they do not go overly past the 80 character mark in length.
- [x] Let's place the BIC formula in the prereq's in a MathJax equation.
- [ ] Use the `validate` and `need` functions on the plots to show helpful feedback on what the user needs to do to have the graphs show up (instead of empty space)
- [x] Replace the clear text place holders (e.g., "We need some text here") with actual descriptive text.
- [x] Expand the opening text on the Overview page; this connects with the prior calls to expand the app's description.
- [ ] Move the aria-label text for the graphs into the new `alt` argument of `renderPlot`
- [ ] There are incorrect usages of `fluidRow` and `column`.
- [x] The "Hints" label is not correctly tagged; these should be inside of `p` and then have the `strong` tag.
- [x] Use meaningful names for objects (e.g., "button" is the name for a checkbox that shows/hides an answer model)
- [x] Use `awesomeCheckbox` instead of `prettyCheckbox` (drop the `icon` argument and include `status = "default"`)
- [x] Make sure to properly format the code and explicitly name function arguments.
- [x] There is a comment in the code on why "Question 7" is being graded. Why is this question being graded? How does the user get anything other than an "incorrect" icon?
- [ ] The "i" button needs proper formatting (missing the icon)
- [ ] References need to be updated.

ymy5178 committed:
CODE CHANGE IN QUESTION 7 AND SOME TEXT CHANGE

CHANGES ARE MADE UNTIL REPLACE THE CLEAR TEXT PLACE HOLDERS.

ymy5178 committed:
things on checklist

Sburke26 reviewed (changes requested):
Some issues I noticed while looking through the app and code:

The year badge in the description file is not updated.
The info button is missing the circled i image.
The app is missing a "cite this app as..." entry.
The data frames are not done before the ui and server definitions.
There is no use of the "alt" argument.
Some instances where camelCase is not used for object naming. And some objects could be named better. 

Suggestions:

The first question is a response question while the rest are drop down choices. Maybe, we should change it to a dropdown as well to maintain a specific style. Also, if there is a way to display models without hitting the refresh data first, it may beneficial to look for a solution to make the model more intuitive.

robchapp reviewed (changes requested):
Similar to previous review. I suggest adding boxes for the prerequisites so it looks better style wise. camelCase is not used throughout the code, and objects could be more descriptive. The date in the description and readme files. Otherwise I think the app runs well

robchapp reviewed (changes requested):
This is very out of the style guide, not having an app.r, a description file, or a docs file. I could not get the app to run locally, and the app is not organized well with an outline. I would suggest scrapping this and re-doing it with the current format.

robchapp committed:
Style Guide Fixes

robchapp committed:
Update app.R

Sburke26 reviewed (changes requested):
Good work with the updates.
Here are a couple of things I noticed while reviewing:

There are a lot of missing whitespace in the code, as indicated by the blue circles with i's in rstudio. 
I  noticed there was a missing equal sign for the eventExpr argument for the refresh in line 603.
There were a couple of console messages stating, "Warning in log(vr) : NaNs produced." 

Stylistically, it may be better to move the prerequisites button on the overview page below the instructions and move the explore button to the prerequisites page instead of having the prerequisites button between instructions. Also, in the Explore page , it may be more appealing to put the user input slider and refresh/generate buttons in a wellpanel.

robchapp committed:
Post Review Update

Sburke26 reviewed (changes requested):
Good work with the updates. The app functions well. There are only a few issues I saw:

There is still a decent amount of spacing issues in the code (mostly just spaces between "){" ).
There are some expressions such as:
" output$aPlot <- renderPlot(
    expr = {
    plot(best.model, scale = "adjr2", main = "Adjusted R-Squared")
  })" 

that would look better with indentation as:

output$aPlot <- renderPlot(
    expr = {
     plot(best.model, scale = "adjr2", main = "Adjusted R-Squared")
   }
) "

there are portions of aria-labels that need to be updated to only use the alt text argument like:
 tags$script(HTML(
                "$(document).ready(function() {
            document.getElementById('Cplot').setAttribute('aria-label',
            `model selection plot under the criterion Mallow's Cp criterion`)
            })"
              )
            )
where there are text styles like bold words, Use tags$strong() and other tags$

I also think it may be better to remove the br() between each instruction.

robchapp committed:
Post-Review Updates

Sburke26 committed:
minor updates

Sburke26 reviewed (changes requested):
Good work on the updates.

I made small adjustments to the spacing and indents to hopefully clear that problem.  I did not realize Github removed the indents from my example in my last review. I would just make sure the updates I made don't clash with anything important.

The only thing I would say to do is make sure that all the rendered plots have alt text. I think the 'Using Real Data' plots don't have alt text.

robchapp committed:
Added alt text

Sburke26 reviewed (dismissed):
The changes look good, and the app functions well. 
The only thing I can think of is to maybe see if you can use an array or some other method to possibly shorten the code for the long chain of conditional statements in some areas. 
Either way, good job.

robchapp committed:
Post Review Updates

neilhatfield committed:
Update app.R

Changes to alter the plotting of the variable selection plots and to streamline some of the code.

robchapp committed:
Final Fixes

robchapp committed:
Create ToDoList.txt

robchapp committed:
Final Updates

robchapp committed:
Update app.R
